dali/pipeline/data/tensor_vector.cc

```diff
@@ -440,10 +586,41 @@ void TensorVector<Backend>::resize_tensors(int new_size) {
         tensors_[i]->Reset();
       }
     }
+    // TODO(klecki): Do not keep the invalidated tensors - this prevents memory hogging but
+    // also gets rid of reserved memory.
+    // tensors_.resize(new_size);
   }
   curr_tensors_size_ = new_size;
 }
 
+template <typename Backend>
+void TensorVector<Backend>::PropagateUp(bool contiguous) {
```

Comment:
PropagateUp... what? What does it mean "up"? This calls for a better name.

Reply:
PropagateUpTheMetadataFromTheIndividuallySetSamplesToTheBatchObject, just like the [Constable Visit](https://wiki.lspace.org/Visit-The-Infidel-With-Explanatory-Pamphlets)

Reply:
Which direction is "Up"? Without your explanation, it's anything but obvious.
How about:
```suggestion
void TensorVector<Backend>::UpdateMetaFromSamples(bool contiguous) {
```
This clearly says that it's
a) metadata
b) from samples
c) that metadata is updated (presumably the main metadata as opposed to per-sample)

Reply:
It's not just the `Meta`, this would be highly misleading. We have a lot of other, proper metadata/properties that are not a part of GetMeta/SetMeta.